[Feature][SQLTask] Suggest SQL task support variables in anywhere

zhuxt2015 opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request
fix #10745
<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

zhuxt2015 committed:
fix

zhuxt2015 committed:
fix

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -291,20 +292,20 @@ private void sendAttachment(int groupId, String title, String content) {
-            ResultSet resultSet = statement.executeQuery();
+    private String executeQuery(Connection connection, String sql, String handlerType) throws Exception {
+        try (Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(sql);
```
Please close the ResultSet

zhuxt2015 committed:
close ResultSet

zhuxt2015 committed:
fix checkstyle

caishunfeng commented:
Hi @zhuxt2015 please resolve the conflicts.

zhuxt2015 committed:
resolve conflicts

zhuxt2015 committed:
remove unuse function

zhuxt2015 committed:
use SQLException instead of Exception

zhuangchong committed:
update PyDolphinScheduler documentation link. (#11474)

zzzhangqi committed:
[doc] Use Rainbond deployment DolphinScheduler Cluster (#10951)

* [doc] Use Rainbond deployment DolphinScheduler Cluster

Signed-off-by: Qi Zhang <smallqi1@163.com>

* Update docs/docs/en/guide/installation/rainbond.md

* Update docs/docs/en/guide/installation/rainbond.md

* Update docs/docs/zh/guide/installation/rainbond.md

* Update docs/docs/zh/guide/installation/rainbond.md

* [doc] modify rainbond doc and add image

Signed-off-by: Qi Zhang <smallqi1@163.com>

* fix: img link

Signed-off-by: Qi Zhang <smallqi1@163.com>

* fix: modify proposal

Signed-off-by: Qi Zhang <smallqi1@163.com>

* docs: move rainbond docs to integration

Signed-off-by: Qi Zhang <smallqi1@163.com>

Signed-off-by: Qi Zhang <smallqi1@163.com>
Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -341,116 +346,6 @@ private void close(Connection connection) {
-     */
-    private PreparedStatement prepareStatementAndBind(Connection connection, SqlBinds sqlBinds) {
-        // is the timeout set
-        boolean timeoutFlag = taskExecutionContext.getTaskTimeoutStrategy() == TaskTimeoutStrategy.FAILED
```
It seems miss the timeout action.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -341,116 +346,6 @@ private void close(Connection connection) {
-
-        //new
-        //replace variable TIME with $[YYYYmmddd...] in sql when history run job and batch complement job
-        sql = ParameterUtils.replaceScheduleTime(sql, taskExecutionContext.getScheduleTime());
-        // special characters need to be escaped, ${} needs to be escaped
-        setSqlParamsMap(sql, rgex, sqlParamsMap, paramsMap,taskExecutionContext.getTaskInstanceId());
-        //Replace the original value in sql ！{...} ，Does not participate in precompilation
-        String rgexo = "['\"]*\\!\\{(.*?)\\}['\"]*";
-        sql = replaceOriginalValue(sql, rgexo, paramsMap);
-        // replace the ${} of the SQL statement with the Placeholder
-        String formatSql = sql.replaceAll(rgex, "?");
-        // Convert the list parameter
-        formatSql = ParameterUtils.expandListParameter(sqlParamsMap, formatSql);
-        sqlBuilder.append(formatSql);
```
Why remove these logic?

caishunfeng commented:
BTW, @zhuxt2015 can you add some UT for it?

EricGao888 committed:
[Doc][Security] Update instructions on worker groups (#11483)

* Update instructions on worker groups

songjianet committed:
[Refactor][UI] Refactor task definition batch stream using NSpace component. (#11485)

* [Refactor][UI] Refactor task definition batch stream using NSpace component.

* [Refactor][UI] Refactor task definition batch stream using NSpace component.

* [Refactor][UI] Refactor task definition batch stream using NSpace component.

songjianet committed:
[Fix][UI] Fix the table header in the workflow instance. (#11487)

songjianet committed:
[Refactor][UI] Refactor task definition version using NSpace component. (#11495)

songjianet committed:
[Doc] Update UI readme. (#11496)

kezhenxu94 committed:
Support reading application properties from kubernetes config map (#11402)

Amy0104 committed:
[Fix][UI] Fix the preTask options are unavailable when creating a task in the task definition page. (#11501)

Commit:
[Feature-10182][task-datax] (#11387)

In the process of realizing datax synchronization, use custom json for better connection to enable kerberos-authenticated clusters, such as using plug-ins such as hdfs and hbase

zhongjiajie committed:
[fix] Database change not sync (#11258)

* [fix] Database change not sync

* Make del sql upgrade same as release version
* Correct data quality dml sql
* Upgrade database DDL to avoid log path too long error

* Add missing if exists

EricGao888 committed:
[Feature][Style] Make sure there is one and only one empty line at the end of each java file (#11520)

guowei-su committed:
[DS-11481][docs] Add log specification document to contribution guidelines. (#11484)

fangyinc committed:
[test-11432][HttpTask] Add unit tests to http task (#11453)

* [test][HttpTask] Add unit tests to http task

* [test][HttpTask] fix http task test code format problem

* [test][HttpTask] fix http task test code replace http code to constants

Zephyrcf committed:
[Improvement] File deletion logic optimization (#11519)

* [Improvement] File deletion logic optimization

Zephyrcf committed:
[Improvement-11468] Support re running historical version workflow (#11489)

* Support re running historical version workflow

 Support re running historical version workflow

* Update ExecutorServiceImpl.java

* Update ExecutorServiceImpl.java

jieguangzhou committed:
[Feature] New task plugin Pytorch (#11498)

Co-authored-by: Eric Gao <ericgao.apache@gmail.com>

Amy0104 committed:
[Fix][UI] Remove unsupported datasource types in the datax task. (#11527)

ruanwenjun committed:
Set process instance and task priority default value as 2 (#11539)

EricGao888 committed:
Enable spotless to fix markdown checkstyle and refactor faq.md as an example (#11384)

106umao committed:
[Improvement]Add a stop function when you are ready to do pause operation (#11543)

* add can stop for execute state machine

* add execute type check for pause operation

huage1994 committed:
[Improvement] [LDAP] Bug fix: Config value return "null" instead of real null currently. (#11325)

ruanwenjun committed:
Remove logger header in task log file (#11555)

itzhang89 committed:
[fix-11404]: make the common.properties to configurable on values.yaml (#11441)

and set the resource.storage.type default to hdfs

fuchanghai committed:
[feature-11408] add hive option (#11493)

kezhenxu94 committed:
Fix some resource leak (#11576)

zhuxt2015 committed:
add query timeout logic
